commands/audit/sca/cocoapods/cocoapods.go

```diff
@@ -0,0 +1,275 @@
+			}
+		}
+	}
+	return podPositions, nil
+}
+
+func FixTechDependency(dependencyName, dependencyVersion, fixVersion string, descriptorPaths ...string) error {
+	for _, descriptorPath := range descriptorPaths {
+		path.Clean(descriptorPath)
+		if !strings.HasSuffix(descriptorPath, "Podfile") {
+			log.Logger.Warn("Cannot support other files besides Podfile: %s", descriptorPath)
+			continue
+		}
+		data, err := os.ReadFile(descriptorPath)
+		var newLines []string
+		if err != nil {
+			continue
+		}
+		lines := strings.Split(string(data), "\n")
+		foundDependency := false
+		var tempIndex int
+		for index, line := range lines {
+			if strings.Contains(line, dependencyName) {
+				foundDependency = true
+				tempIndex = index
+			}
+			// This means we are in a new dependency (we cannot find dependency name and version together)
+			if index > tempIndex && foundDependency && strings.Contains(line, "pod") {
+				foundDependency = false
+			} else if foundDependency && strings.Contains(line, dependencyVersion) {
+				newLine := strings.Replace(line, dependencyVersion, fixVersion, 1)
+				newLines = append(newLines, newLine)
+				foundDependency = false
+			} else {
+				newLines = append(newLines, line)
+			}
+		}
+		output := strings.Join(newLines, "\n")
+		err = os.WriteFile(descriptorPath, []byte(output), 0644)
+		if err != nil {
+			return fmt.Errorf("failed to write file: %v", err)
+		}
+	}
+	return nil
+}
+
+func GetPackageName(longPkgName string) string {
+	if strings.Contains(longPkgName, "/") {
+		splitNameParts := strings.Split(longPkgName, "/")
+		longPkgName = splitNameParts[0]
+	}
+	return longPkgName
+}
+
+func GetPodDependenciesGraph(data string) (map[string][]string, map[string]string) {
+	var currentMainDep string
+	lines := strings.Split(data, "\n")
+	dependencyMap := make(map[string][]string, len(lines))
+	versionMap := make(map[string]string, len(lines))
+	for _, line := range lines {
+		line = strings.ReplaceAll(line, "\"", "")
+		mainDepMatch := mainDepRegex.FindStringSubmatch(line)
+		if len(mainDepMatch) == 3 {
+			versionMatch := versionRegex.FindStringSubmatch(line)
+			currentMainDep = GetPackageName(mainDepMatch[1])
+			_, ok := dependencyMap[currentMainDep]
+			if !ok {
+				dependencyMap[currentMainDep] = []string{}
+				versionMap[currentMainDep] = versionMatch[1]
+			}
+			continue
+		}
+		subDepMatch := subDepRegex.FindStringSubmatch(line)
+		if len(subDepMatch) == 2 && currentMainDep != "" {
+			subDependency := subDepMatch[1]
+			if subDependency == GetPackageName(subDependency) {
+				dependencyMap[currentMainDep] = append(dependencyMap[currentMainDep], subDependency)
+			}
+		}
+	}
+	return dependencyMap, versionMap
+}
+
+func extractPodsSection(filePath string) (string, error) {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+	content := string(data)
+	startIndex := strings.Index(content, "PODS:")
+	if startIndex == -1 {
+		return "", fmt.Errorf("PODS: section not found")
```

Comment:
is `PODS` section required for all cocoapods projects? if it is possiable to have a project like it, we will alway return error on the project with no dependencies

Reply:
yes its part of the Podfile.lock syntax